codes/c/chapter_divide_and_conquer/build_tree.c

```diff
@@ -0,0 +1,62 @@
+/**
+  * File : build_tree.c
+  * Created Time: 2023-10-07
+  * Author : lucas (superrat6@gmail.com)
+  */
+
+#include "../utils/common.h"
+
+/* 构建二叉树：分治 */
+TreeNode* dfs(int* preorder, int* inorder, int i, int l, int r, int size) {
+    // 子树区间为空时终止
+    if (r - l < 0)
+        return NULL;
+    // 初始化根节点
+    TreeNode* root = (TreeNode*)malloc(sizeof(TreeNode));
+    root->val = preorder[i];
+    root->left = NULL;
+    root->right = NULL;
+    // 查询 m ，从而划分左右子树
+    int m;
+    for (m = l; m <= r; m++) {
+        if (inorder[m] == preorder[i])
```

Comment:
Please use a hashtable named as `inorderMap` to reduce the time complexity to O(1)